python_modules/dagster/dagster/_core/definitions/unresolved_asset_sensor_definition.py

```diff
@@ -0,0 +1,279 @@
+
+        The cursor for this sensor is a dictionary mapping stringified AssetKeys to dictionaries. The
+        dictionary for an asset is a mapping of the stringified AssetKeys of its parents to the storage id
+        of the latest materialization that has resulted in the child asset materializing
+        """
+
+        def sensor_fn(context):
+
+            cursor_dict = json.loads(context.cursor) if context.cursor else {}
+            should_materialize = []
+            cursor_update_dict = {}
+
+            # sort the assets topologically so that we process them in order
+            toposort_assets = list(toposort.toposort(upstream))
+            # unpack the list of sets into a list
+            toposort_assets = [
+                asset for layer in toposort_assets for asset in layer if asset in upstream.keys()
+            ]
+
+            # determine which assets should materialize based on the materialization status of their
+            # parents
+            for a in toposort_assets:
+                a_cursor = self._get_cursor(cursor_dict, a)
+                parent_materialization_records = self._get_parent_materializations(
+                    context,
+                    parent_assets=upstream[a],
+                    cursor=a_cursor,
+                    will_materialize_list=should_materialize,
+                )
+
+                if all(
+                    [
+                        materialization_status
+                        for materialization_status, _ in parent_materialization_records.values()
+                    ]
+                ):  # TODO should we allow the user to specify all() vs any() here?
+                    should_materialize.append(a)
+                    cursor_update_dict[str(a)] = self._make_cursor_update_dict(
+                        parent_event_records=parent_materialization_records, current_cursor=a_cursor
+                    )
+                else:
+                    # if the asset shouldn't be materialized, then keep the cursor the same
+                    cursor_update_dict[str(a)] = self._get_cursor(cursor_dict, a)
+
+            if len(should_materialize) > 0:
+                context.update_cursor(json.dumps(cursor_update_dict))
+                return RunRequest(run_key=f"{context.cursor}", asset_selection=should_materialize)
+
+        return SensorDefinition(
+            evaluation_fn=sensor_fn,
+            name=self.name,
+            job_name="__ASSET_JOB",
+            minimum_interval_seconds=self._minimum_interval_seconds,
+        )
+
+    def resolve(
+        self,
+        assets: Sequence["AssetsDefinition"],
+        source_assets: Sequence["SourceAsset"],
+    ) -> SensorDefinition:
+        self._source_asset_keys = [asset.key for asset in source_assets]
+        return self._make_sensor(self._get_upstream_mapping(assets, source_assets))
+
+
+@experimental
+def build_asset_sensor(
+    selection: AssetSelection,
+    name: str,
```

Comment:
maybe all of the assets?

Reply:
that could get big in some cases - maybe if it gets really big, truncate it and include a hash of the remaining assets?

Reply:
i originally misread the comment, and thought you were suggesting a default for the `selection` param (hence, my original suggestion that the default be all assets). updated to reflect a default name for the sensor. do you think we should be generating a unique default name so that if someone creates multiple of these sensors they don't have to provide names for any of them?